packages/api/lib/ems.js

```diff
@@ -0,0 +1,280 @@
+ * process.env.ems_provider: default to 'cumulus'
+ * process.env.stackName
+ * process.env.bucket
+ */
+
+/**
+ * For each EMS report type (ingest, archive, delete),
+ * map the EMS fields to CUMULUS granule/deletedgranule record fields,
+ */
+const emsMappings = {
+  ingest: {
+    dbID: 'granuleId',
+    product: 'collectionId', // shortName part
+    versionID: 'collectionId', // versionID part
+    productVolume: 'productVolume',
+    productState: 'status',
+    externalDataProvider: 'provider',
+    processingStartDateTime: 'processingStartDateTime',
+    processingEndDateTime: 'processingEndDateTime',
+    timeToArchive: 'timeToArchive',
+    timeToPreprocess: 'timeToPreprocess',
+    timeToXfer: 'duration'
+  },
+
+  archive: {
+    dbID: 'granuleId',
+    product: 'collectionId',
+    productVolume: 'productVolume',
+    totalFiles: 'files', // total # files
+    insertTime: 'createdAt',
+    beginningDateTime: 'beginningDateTime',
+    endingDateTime: 'endingDateTime',
+    productionDateTime: 'productionDateTime',
+    localGranuleID: 'granuleId',
+    versionID: 'collectionId',
+    deleteFromArchive: 'deleteFromArchive', // N
+    deleteEffectiveDate: 'deleteEffectiveDate', // null
+    lastUpdate: 'lastUpdateDateTime'
+  },
+
+  delete: {
+    dbID: 'granuleId',
+    deleteEffectiveDate: 'deletedAt'
+  }
+};
+
+/**
+ * build and elasticsearch query parameters
+ *
+ * @param {string} esIndex - es index to search on
+ * @param {string} type - es document type to search on
+ * @param {string} startTime - startTime of the records
+ * @param {string} endTime - endTime of the records
+ * @returns {Object} query parameters
+ */
+function buildSearchQuery(esIndex, type, startTime, endTime) {
+  const timeFieldName = (type === 'granule') ? 'createdAt' : 'deletedAt';
+  const params = {
+    index: esIndex,
+    type: type,
+    scroll: '30s',
+    size: process.env.ES_SCROLL_SIZE || 1000,
+    body: {
+      query: {
+        bool: {
+          must: [
+            {
+              range: {
+                [`${timeFieldName}`]: {
+                  gte: moment.utc(startTime).toDate().getTime(),
+                  lt: moment.utc(endTime).toDate().getTime()
+                }
+              }
+            },
+            {
+              terms: {
+                // filter out 'running' status
+                status: ['failed', 'completed']
+              }
+            }]
+        }
+      }
+    }
+  };
+  if (type === 'deletedgranule') params._source = ['granuleId', 'deletedAt'];
+  return params;
+}
+
+/**
+ * get the value of EMS record field from corresponding field of the  granule record
+ *
+ * @param {Object} granule - es granule record
+ * @param {string} emsField - EMS field
+ * @param {string} granField - granule field
+ * @returns {string} granule metadata for EMS
+ */
+function getEmsFieldFromGranField(granule, emsField, granField) {
+  const metadata = get(granule, granField, null);
```

Comment:
There's no need to use `get()` here, you can just say `const metadata = granule[granField];`